Gems/SurfaceData/Code/Source/TerrainSurfaceDataSystemComponent.h

```diff
@@ -58,25 +56,19 @@ namespace SurfaceData
 
         //////////////////////////////////////////////////////////////////////////
         // SurfaceDataProviderRequestBus
-        void GetSurfacePoints(const AZ::Vector3& inPosition, SurfacePointList& surfacePointList) const;
-
-        ////////////////////////////////////////////////////////////////////////////
-        // CrySystemEvents
-        void OnCrySystemInitialized(ISystem& system, const SSystemInitParams& systemInitParams) override;
-        void OnCrySystemShutdown(ISystem& system) override;
+        void GetSurfacePoints(const AZ::Vector3& inPosition, SurfaceData::SurfacePointList& surfacePointList) const;
```

Comment:
I'm a little confused about how you could get multiple points back from one point in. Is this for overlapping terrain? Would there be a different function for getting the final terrain?

Reply:
The API is defined by the SurfaceData system and is used for more than just terrain.  For any given surface type (shape, mesh, terrain, etc), the API shouldn't restrict the possibility of returning multiple vertically-overlapping points.  For example, a sphere could return both the top and the bottom of the sphere at a given XY location.